src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Button from '@ecl/ec-react-component-button';
+
+import TimelineItem from './Timeline2Item';
+
+const Timeline = ({
+  items,
+  toggleCollapsed,
+  toggleExpanded,
+  button,
+  className,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-timeline2');
+
+  // Calculate collapse threshold
+  const threshold =
+    items.length > 12
+      ? Math.min(Math.floor(items.length * 0.2), 7)
+      : items.length;
+  const hidenItemsLength = items.length - threshold - 2;
+
+  const toggleLabelCollapsed = toggleCollapsed
```

Comment:
wondering if this logic is necessary or if it's up to the developer/consumer to implement it

when the data is available, the developer "knows" how many items there are, he could update the label on his side before feeding the component

Reply:
well, he knows how many items there are in total, yes. But he may not be aware of the calculus to get the collapse/expand thershold. So he doesn't know how many are hidden

Reply:
I understand your point of view. In the end, it depends on whether or not the developer has to implement the calculus or not. Let's close this discussion and continue on the one above ⬆️